internal/index_read.go

```diff
@@ -102,26 +103,19 @@ func readBins(r io.Reader, typ string) ([]Bin, *ReferenceStats, error) {
 	return bins, stats, nil
 }
 
-func readChunks(r io.Reader, n int32, typ string) ([]bgzf.Chunk, error) {
+func readChunks(r io.Reader, n int32, typ string, buf [16]byte) ([]bgzf.Chunk, error) {
```

Comment:
If this code stays the way it is, it needs a comment explaining why; semantically it makes no sense since the buf is passed in zeroed and is copied each call (when the holiday is over, I might ask why the performance difference is seen).

What is the performance difference between `var buf [16]byte` and as it is currently?

Have you profiled why the cost is different?